### PR TITLE
internal/actor: migrate middleware to use new logger

### PR DESCRIPTION
**Pull request by davejrt**
migrate HTTPMiddleware from log15 to `sourcegraph/log` as part of 
https://github.com/sourcegraph/sourcegraph/issues/40896

## Test plan

green tests on this PR

### CODE REVIEW

File: internal/actor/http_test.go [hunk 1]
```diff
@@ -114,8 +115,9 @@ func TestHTTPMiddleware(t *testing.T) {
 		wantActor: &Actor{AnonymousUID: "anonymousUID"},
 	}}
 	for _, tt := range tests {
+		logger := logtest.Scoped(t)
 		t.Run(tt.name, func(t *testing.T) {
-			handler := HTTPMiddleware(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+			handler := HTTPMiddleware(logger, http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
```
Comment (bobheadxi) [inline internal/actor/http_test.go:120]:
```suggestion
			handler := HTTPMiddleware(logtest.Scoped(t), http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
```

File: internal/actor/http.go [hunk 2]
```diff
@@ -130,9 +130,9 @@ func HTTPMiddleware(next http.Handler) http.Handler {
 		default:
 			uid, err := strconv.Atoi(uidStr)
 			if err != nil {
-				log15.Warn("invalid user ID in request",
-					"error", err,
-					"uid", uidStr)
+				logger.Warn("invalid user ID in request",
```
Comment (bobheadxi) [inline internal/actor/http.go:133]: We can add `WithTrace` here as well, HTTP requests are often traced

File: internal/actor/http_test.go (hunk 1 above)
Comment (davejrt) [reply internal/actor/http_test.go:120]: Thanks, I think this was a case of just making things not red and being on auto pilot

File: internal/actor/http.go [hunk 3]
```diff
@@ -102,7 +102,7 @@ func (t *HTTPTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 // 🚨 SECURITY: This should *never* be called to wrap externally accessible handlers (i.e.
 // only use for internal endpoints), because internal requests can bypass repository
 // permissions checks.
-func HTTPMiddleware(next http.Handler) http.Handler {
+func HTTPMiddleware(logger log.Logger, next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
```
Comment (bobheadxi) [inline internal/actor/http.go:106]: We might want to sub-scope this:

```suggestion
func HTTPMiddleware(logger log.Logger, next http.Handler) http.Handler {
    logger = logger.Scoped("actors", "middleware for parsing actor from HTTP headers")

	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
```

Comment (davejrt) [reply internal/actor/http.go:106]: TIL

File: internal/actor/http.go (hunk 2 above)
Comment (davejrt) [reply internal/actor/http.go:134]: In addition to, or as part of. I couldn't find any good examples to go off so I'm not sure if my latest commit is what you meant here.

Comment (davejrt) [reply internal/actor/http.go:134]: https://github.com/sourcegraph/sourcegraph/pull/41336/commits/8c029e0b9aab9708bd769710a667906be4a21057

File: internal/actor/http.go [hunk 4]
```diff
@@ -134,6 +135,7 @@ func HTTPMiddleware(logger log.Logger, next http.Handler) http.Handler {
 					log.Error(err),
 					log.String("uid", uidStr))
 				metricIncomingActors.WithLabelValues(metricActorTypeInvalid, path).Inc()
+				logger.WithTrace(trace.Context(ctx))
```
Comment (bobheadxi) [inline internal/actor/http.go:138]: Logger functions never mutate the underlying logger - you want something like:

```go
logger.WithTrace(trace.Context(ctx)).
    Warn("invalid user ID in request",
		log.Error(err),
		log.String("uid", uidStr))
```

Comment (bobheadxi) [reply internal/actor/http.go:138]: You can also use `trace.Logger`: https://docs.sourcegraph.com/dev/how-to/add_logging#traced-sub-loggers

Comment (davejrt) [reply internal/actor/http.go:138]: our logging is like an onion...just peeling back the layers

Comment (bobheadxi) [reply internal/actor/http.go]: I've opened a PR here to try and include more examples: https://github.com/sourcegraph/sourcegraph/pull/41384

**Review by bobheadxi — APPROVED** [review]
nice 😁

**Review by jhchabran — APPROVED** [review]

**Pull request merged**